tests/subsys/dsp/basicmath/src/f32.c

```diff
@@ -219,30 +256,27 @@ DEFINE_TEST_VARIANT4(basic_math_f32, zdsp_scale_f32, 0p5_11, in_com1, 0.5f, ref_
 DEFINE_TEST_VARIANT4(basic_math_f32, zdsp_scale_f32, long, in_com1, 0.5f, ref_scale,
 		     ARRAY_SIZE(in_com1));
 
-static void test_zdsp_dot_prod_f32(
-	const uint32_t *input1, const uint32_t *input2, const uint32_t *ref,
-	size_t length)
+static void test_zdsp_dot_prod_f32(const DSP_DATA uint32_t *input1, const DSP_DATA uint32_t *input2,
+				const uint32_t *ref, size_t length)
 {
-	float32_t *output;
+	DSP_DATA float32_t *output;
 
 	/* Allocate output buffer */
-	output = malloc(1 * sizeof(float32_t));
+	output = (DSP_DATA float32_t *)malloc(1 * sizeof(float32_t));
```

Comment:
Do we really need the attributes specified by the `DSP_DATA` for the type cast here?

Reply:
It's needed for mwdt compiler.